test/add-rm-pkg-deps.js

```diff
@@ -122,6 +122,56 @@ t.test('add', t => {
     devDependencies: { foo: '2' },
   }, 'update dev')
 
+  t.strictSame(add({
+    pkg: {
+      devDependencies: { foo: '1' },
+      peerDependencies: { foo: '1' },
+    },
+    add: [ foo2 ],
+  }), {
+    devDependencies: { foo: '2' },
+    peerDependencies: { foo: '2' },
+  }, 'update dev and peer together')
+
+  t.strictSame(add({
+    pkg: {
+      devDependencies: { foo: '1' },
+      peerDependencies: { foo: '1' },
+      peerDependenciesMeta: { foo: { optional: true }},
+    },
+    add: [ foo2 ],
+  }), {
+    devDependencies: { foo: '2' },
+    peerDependencies: { foo: '2' },
+    peerDependenciesMeta: { foo: { optional: true }},
+  }, 'update dev and peerOptional together')
+
+  t.strictSame(add({
+    pkg: {
+      devDependencies: { foo: '*' },
+      peerDependencies: { foo: '1' },
+      peerDependenciesMeta: { foo: { optional: true }},
+    },
+    add: [ foo2 ],
+  }), {
+    devDependencies: { foo: '2' },
+    peerDependencies: { foo: '2' },
+    peerDependenciesMeta: { foo: { optional: true }},
+  }, 'update dev and peerOptional together')
+
+  t.strictSame(add({
+    pkg: {
+      devDependencies: { foo: '1' },
+      peerDependencies: { foo: '*' },
+      peerDependenciesMeta: { foo: { optional: true }},
+    },
+    add: [ foo2 ],
+  }), {
+    devDependencies: { foo: '2' },
+    peerDependencies: { foo: '2' },
```

Comment:
i would expect a peer of "*" to always remain a peer of "*", otherwise this creates a potentially unintentional breaking change.

Reply:
That's a fair suggestion, but it's not how it works today with `--save-peer`, and changing the status quo there will require some deeper analysis.